src/node/history.h

```diff
@@ -403,14 +424,33 @@ namespace ccf
       return tree.get_root();
     }
 
-    void append(const std::vector<uint8_t>& data) override
+    crypto::Sha256Hash get_r_root() override
+    {
+      return r_tree.get_root();
+    }
+
+    void append(
+      const std::vector<uint8_t>& replicated,
+      const std::vector<uint8_t>& all_data) override
     {
-      append(data.data(), data.size());
+      append(
+        replicated.data(), replicated.size(), all_data.data(), all_data.size());
     }
 
-    void append(const uint8_t* data, size_t size) override
+    void append(
+      const uint8_t* replicated,
+      size_t replicated_size,
+      const uint8_t* all_data,
+      size_t all_data_size) override
     {
-      crypto::Sha256Hash h({{data, size}});
+      if (update_r_tree())
```

Comment:
Minor: for all the other history calls (e.g. `rollback()`), the replicated tree is updated _after_ the other tree is updated. Here however, it is updated first. We should change the order to make it consistent.